### PR TITLE
feat: Add question ID to response from Swagger API

### PR DESCRIPTION
**Pull request by frostyfrog**
When a question is asked and an ID is generated, that ID is not
communicated to the user. One of the requests that has come up is for
that ID to be returned in the response instead of just a "success"
message. While making this change, I also noticed that there was an
instance where we could fail to send & store the question but still
report a success state. With a little refactoring, this problem has also
been fixed and the code made a little cleaner.

Signed-off-by: Colton Wolkins (Indicio work address) <colton@indicio.tech>

### CODE REVIEW

**Review by cjhowland — APPROVED** [review]

**Comment by frostyfrog** [comment]
Thank you @cjhowland 😄

File: acapy_plugin_qa/v1_0/routes.py [hunk 1]
```diff
@@ -153,28 +154,35 @@ async def send_question(request: web.BaseRequest):
...
+    async with context.session() as session:
+        await record.save(session, reason="New question received")
 
-    return web.json_response({"success": True})
+    await outbound_handler(msg, connection_id=connection_id)
+
+    return web.json_response(
+        {
+            "state": "QUESTION_SENT",
+            "question_id": msg._id,
```
Comment (reflectivedevelopment) [inline acapy_plugin_qa/v1_0/routes.py:183]: question_id should be thread_id to have consistent results.

File: acapy_plugin_qa/v1_0/routes.py [hunk 2]
```diff
@@ -153,28 +154,35 @@ async def send_question(request: web.BaseRequest):
...
+    )
+    async with context.session() as session:
+        await record.save(session, reason="New question received")
 
-    return web.json_response({"success": True})
+    await outbound_handler(msg, connection_id=connection_id)
+
+    return web.json_response(
+        {
+            "state": "QUESTION_SENT",
```
Comment (reflectivedevelopment) [inline acapy_plugin_qa/v1_0/routes.py:182]: State should match previous state.

**Review by reflectivedevelopment — COMMENTED** [review]
Maybe we should match the previous pattern of /issue-credential/create where they return the credential_exchange record.

Does it make sense to use thread_id for the questions, or add our own id field like question_id, even if we use the thread_id for the question_id field?

**Pull request merged**